mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulPrepareGameScreenTest.kt

```diff
@@ -80,41 +93,192 @@ class StatefulPrepareGameScreenTest {
+    rule.onNodeWithText("user2").performClick()
+    rule.onNodeWithText(strings.prepareGamePlay).performClick()
+
+    val fetchedMatch = chess.matches(currentUser).mapNotNull { it.firstOrNull() }.first()
+    assertThat(fetchedMatch.white.filterNotNull().first().uid).isEqualTo(user2.uid)
+    assertThat(fetchedMatch.black.filterNotNull().first().uid).isEqualTo(currentUser.uid)
+  }
+
+  @Test
+  fun unfollowingAUser_DoesNotDisplayItAsProfileAnymore() = runTest {
+    val auth = emptyAuth()
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(name = "user2")) }
+      collection("games") {}
+    }
+    val facade = AuthenticationFacade(auth, store)
+    val social = SocialFacade(auth, store)
+    val chess = ChessFacade(auth, store)
+
+    facade.signUpWithEmail("user1@email", "user1", "password")
+    val authUser1 = facade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+    val user2 =
+        social.profile(uid = "userId2", user = authUser1).filterIsInstance<Profile>().first()
+    authUser1.follow(user2)
+
+    rule.setContentWithLocalizedStrings {
+      ProvideFacades(facade, social, chess) {
+        StatefulPrepareGameScreen(user = authUser1, navigateToGame = {}, cancelClick = {})
+      }
+    }
+
+    rule.onNodeWithText("user2").assertExists()
+    authUser1.unfollow(user2)
+    rule.onNodeWithText("user2").assertDoesNotExist()
+  }
+
+  @Test
+  fun creatingGameFromPrepareGame_callsCallback() = runTest {
+    val auth = emptyAuth()
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(name = "user2")) }
+      collection("games") {}
+    }
+    val facade = AuthenticationFacade(auth, store)
+    val social = SocialFacade(auth, store)
+    val chess = ChessFacade(auth, store)
+
+    facade.signUpWithEmail("user1@email", "user1", "password")
+    val currentUser = facade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+    val user2 =
+        social.profile(uid = "userId2", user = currentUser).filterIsInstance<Profile>().first()
+    currentUser.follow(user2)
+
+    val channel = Channel<Unit>(capacity = 1)
+    val strings =
+        rule.setContentWithLocalizedStrings {
+          ProvideFacades(facade, social, chess) {
+            StatefulPrepareGameScreen(
+                user = currentUser,
+                navigateToGame = { _ ->
+                  channel.trySend(Unit)
+                  channel.close()
+                },
+                cancelClick = {})
+          }
+        }
+
+    rule.onNodeWithText("user2").performClick()
+    rule.onNodeWithText(strings.prepareGamePlay).performClick()
+
+    assertThat(channel.receive()).isEqualTo(Unit)
+  }
+
+  @Test
+  fun cancelingPrepareGame_callsCallback() = runTest {
+    val auth = emptyAuth()
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(name = "user2")) }
+      collection("games") {}
```

Comment:
```suggestion
```

Reply:
Fixed in [9bcca9f](https://github.com/epfl-SDP/android/pull/200/commits/9bcca9f97a6022919e149265ea78cf9f4b472043)